### ISSUE TITLE
In the samples, need to call `clReleaseEvent` to prevent leaks

### ISSUE THREAD

**Issue by hughperkins** [clMathLibraries/clBLAS#167]
In the samples, need to call `clReleaseEvent`, after waiting on it, to prevent memory leaks.  ie:

```
        err = clWaitForEvents(1, &event);
        clReleaseEvent(event);
```

**Comment by TimmyLiu** [comment]
should be fixed by pr #178 . Can we close it?

**Comment by hughperkins** [comment]
Great!  Thank you Timmy.
